### PR TITLE
UCP/RMA: Define protocols from RMA and AMO

### PR DESCRIPTION
**Pull request by yosefe**
Mostly moving code around

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7494/ for details (Mellanox internal link).

File: src/ucp/rma/amo_send.c [hunk 1]
```diff
@@ -0,0 +1,269 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
```
Comment (brminich) [inline src/ucp/rma/amo_send.c]: minor: copyright

File: src/ucp/rma/rma_basic.c [hunk 2]
```diff
@@ -1,130 +1,16 @@
 /**
-* Copyright (C) Mellanox Technologies Ltd. 2001-2018.  ALL RIGHTS RESERVED.
+* Copyright (C) Mellanox Technologies Ltd. 2001-2015.  ALL RIGHTS RESERVED.
```
Comment (brminich) [inline src/ucp/rma/rma_basic.c]: why need to change it back?

Comment (yosefe) [reply src/ucp/rma/rma_basic.c]: rebase issue

**Comment by yosefe** [comment]
@brminich when adding active-message based rma/amo protocols, i want to separate as follows:
rma_send.c - implementation of API calls
rma_basic.c - progress protocol over UCT RMA
rma_sw.c - progress protocol over active messages
(and same for amo)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7501/ for details (Mellanox internal link).

**Pull request merged**